pkg/libzmix/bbs/bbs.go

```diff
@@ -0,0 +1,709 @@
+	if code != 0 {
+		return nil, errors.Errorf("error occurred finishing BBS create proof context: %s", C.GoString(err.message))
+	}
+
+	return &Proof{Buffer: proof}, nil
+}
+
+//Free deallocates the proof instance
+func (r *Proof) Free() {
+	C.bbs_byte_buffer_free(r.Buffer)
+}
+
+type VerifyProofContext Handle
+
+//VerifyProofContextInit initializes a VerifyProofContext
+func VerifyProofContextInit() (VerifyProofContext, error) {
+	var err C.ExternError
+
+	handle := C.bbs_verify_proof_context_init(&err)
+	if err.code != 0 {
+		return VerifyProofContext{}, errors.Errorf("error occurred initializing BBS verify proof context: %s", C.GoString(err.message))
+	}
+
+	return VerifyProofContext{data: handle}, nil
+}
+
+//AddMessages adds the message and the associated index to the VerifyProofContext
+func (r *VerifyProofContext) AddMessage(message string, index int) error {
+	var err C.ExternError
+	var cMessage C.ByteBuffer
+	defer C.bbs_byte_buffer_free(cMessage)
+
+	cMessage.len = (C.long)(len(message))
+	cm := C.CString(message)
+	cMessage.data = (*C.uchar)(unsafe.Pointer(cm))
+	code := C.bbs_verify_proof_context_add_message_bytes(r.data, (C.uint)(index), cMessage, &err)
+	if code != 0 {
+		return errors.Errorf("error occurred adding message bytes to BBS verify proof context: %s", C.GoString(err.message))
+	}
+
+	return nil
+}
+
+//SetProof sets the proof on the VerifyProofContext
+func (r *VerifyProofContext) SetProof(proof HandleByteBuffer) error {
+	var err C.ExternError
+
+	code := C.bbs_verify_proof_context_set_proof(r.data, proof.Buffer, &err)
+	if code != 0 {
+		return errors.Errorf("error occurred setting proof to BBS verify proof context: %s", C.GoString(err.message))
+	}
+
+	return nil
+}
+
+//SetPublicKey sets the bbs public key on the VerifyProofContext
+func (r *VerifyProofContext) SetPublicKey(publicKey HandleByteBuffer) error {
+	var err C.ExternError
+
+	code := C.bbs_verify_proof_context_set_public_key(r.data, publicKey.Buffer, &err)
+	if code != 0 {
+		return errors.Errorf("error occurred setting secret key to BBS create proof context: %s", C.GoString(err.message))
+	}
+
+	return nil
+}
+
+//SetNonce sets the nonce on the VerifyProofContext
+func (r *VerifyProofContext) SetNonce(nonce []byte) error {
+	var err C.ExternError
+	var nonceBuffer C.ByteBuffer
+	defer C.bbs_byte_buffer_free(nonceBuffer)
+
+	nonceBuffer.len = (C.long)(len(nonce))
+	nonceBuffer.data = (*C.uchar)(unsafe.Pointer(&nonce))
+
+	code := C.bbs_verify_proof_context_set_nonce_bytes(r.data, nonceBuffer, &err)
+	if code != 0 {
+		return errors.Errorf("error occurred setting nonce bytes on BBS create proof context: %s", C.GoString(err.message))
+	}
+
+	return nil
+}
+
+//Finish verifies a proof
+func (r *VerifyProofContext) Finish() error {
+	var err C.ExternError
+
+	code := C.bbs_verify_proof_context_finish(r.data, &err)
+	if code != 200 {
+		return errors.Errorf("error occurred finishing BBS verify proof context: %s", C.GoString(err.message))
+	}
```

Comment:
I think this is the method that doesn't return an `err.message` when it fails, it returns a code value.  Can you verify in the Rust code and tailor error messages based on return code?  If not, ignore me!